core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -298,7 +299,28 @@ default Map<String, MountPointInfo> getMountPointInfoSummary() {
    * @throws AccessControlException if permission checking fails
    * @throws InvalidPathException if the path is invalid
    */
-  void delete(AlluxioURI path, DeleteContext context)
+  default void delete(AlluxioURI path, DeleteContext context)
+      throws IOException, FileDoesNotExistException, DirectoryNotEmptyException,
+      InvalidPathException, AccessControlException {
+    try (JournalContext journalContext = createJournalContext()) {
+      delete(path, context, journalContext);
+    }
+  }
+
+  /**
+   * Deletes a given path.
+   * <p>
+   * This operation requires user to have WRITE permission on the parent of the path.
+   *
+   * @param path the path to delete
+   * @param context method context
+   * @param journalContext context for journaling
+   * @throws DirectoryNotEmptyException if recursive is false and the file is a nonempty directory
+   * @throws FileDoesNotExistException if the file does not exist
+   * @throws AccessControlException if permission checking fails
+   * @throws InvalidPathException if the path is invalid
+   */
+  void delete(AlluxioURI path, DeleteContext context, JournalContext journalContext)
```

Comment:
todo: https://github.com/Alluxio/alluxio/pull/17032#discussion_r1131394449 this may work only in EJ